services/w3c/w3c-validation-helper.js

```diff
@@ -0,0 +1,157 @@
+'use strict'
+
+const html5Expression =
+  '^HTML\\s?,\\s?SVG\\s?1\\.1\\s?,\\s?MathML\\s?3\\.0(\\s?,\\s?((ITS\\s?2\\.0)|(RDFa\\s?Lite\\s?1\\.1)))?$'
+const html4Expression =
+  '^HTML\\s?4\\.01\\s?(Strict|Transitional|Frameset)\\s?,\\s?URL\\s?\\/\\s?XHTML\\s?1\\.0\\s?(Strict|Transitional|Frameset)\\s?,\\s?URL$'
+const xhtmlExpression =
+  '^(XHTML\\s?,\\s?SVG\\s?1\\.1\\s?,\\s?MathML\\s?3\\.0(\\s?,\\s?RDFa\\s?Lite\\s?1\\.1)?)|(XHTML\\s?1\\.0\\s?Strict\\s?,\\s?URL\\s?,\\s?Ruby\\s?,\\s?SVG\\s?1\\.1\\s?,\\s?MathML\\s?3\\.0)$'
+const svgExpression =
+  '^SVG\\s?1\\.1\\s?,\\s?URL\\s?,\\s?XHTML\\s?,\\s?MathML\\s?3\\.0$'
+const presetRegex = new RegExp(
+  `(${html5Expression})|(${html4Expression})|(${xhtmlExpression})|(${svgExpression})`,
+  'i'
+)
```

Comment:
I haven't tested these out to confirm, but they appear to be case sensitive. Is that by design? Does the W3C Checker API require case-correctness for the preset param?

Reply:
The API takes a list of space delimited urls representing the schema. An example is     `http://s.validator.nu/svg-xhtml5-rdf-mathml.rnc, http://s.validator.nu/html5/assertions.sch, http://c.validator.nu/all/`. This seems too much to configure for the badge. So I took the preset concept from the GUI version of the validator that maps the presets to the string of schema urls. They are using a Dropdown so I guess there is no casing issue for them. For the Regex, I am using the `i` flag that ignores casing. The following are unit test cases for the preset.
```
      { preset: 'html,svg 1.1,mathml 3.0', expectedResult: true },
      { preset: 'HTML,SVG 1.1,MathML 3.0', expectedResult: true },
```
The only difference in my implementation from the GUI version is instead of doing `HTML + SVG 1.1 + MathML 3.0` I replaced the `+` with the `,`. As `+` has a special meaning in an url and I was having a hard time fighting to get that work well in both the testing and running the dialog help and also the actual badge. The other thing is that for the spaces surrounding `,` I made them optional. So the version we are using is a lot more forgiving.


Screen shots of the GUI can be found at this comment 
> Hey @calebcartwright,
> 
> That is right, as I was working on it. That was my thought about it. But for schema, the validator is actually using a preset. Then that actually selects a set of schemas to validate against.
> 
> Presets and Schema
> ![Screen Shot 2019-10-07 at 3 27 14 PM](https://user-images.githubusercontent.com/35116035/66343173-0c389600-e919-11e9-8266-555c705ab341.png)
> <img alt="Screen Shot 2019-10-07 at 3 40 41 PM" width="930" src="https://user-images.githubusercontent.com/35116035/66343180-1064b380-e919-11e9-8789-da34b76aa181.png">
> 
> Parser
> ![Screen Shot 2019-10-07 at 3 27 22 PM](https://user-images.githubusercontent.com/35116035/66343205-1f4b6600-e919-11e9-9a36-f2704648da63.png)
> 
> Thanks

Reply:
> So the version we are using is a lot more forgiving.

Oops! That's my fault for not reading down far enough. That's also great that we're more forgiving (I wanted to make sure we weren't being _more_ strict) 👍

Reply:
No problem. That was a long one.